### DISCUSSION TITLE
Questions regarding dGPU behaviour

### DISCUSSION THREAD

**Discussion by Beliroz** [seerge/g-helper#2626] [Q&A] [answered]
Hi! Sorry if this question has been asked and answered before. I am new to github and don't know how to find that out. I recently found out about g-helper and installed it. It's much better than AC, for sure, but I have some questions regarding dGPU behavior.

With AC the dGPU would be off by default when running on battery or USB power, and turn itself on when i started programs like Adobe Bridge (which I use quite frequently). When the dGPU activated it would also show "1 Display running on GPU". 
Now it shows 0 displays running on GPU when set to Standard and Eco mode. Hardware acceleration seems to be disabled as well, because photo editing programs run horrendously slow.
The tray icon for the GPU seems to indicate that the GPU is always activated, even in Eco mode.
The tray icon detects the display when switching to Ultimate mode, hardware acceleration works, and i get the option to overclock the display to 165 Hz (which is disabled in Standard and Eco mode).

I have a G14 GA402NI with an RTX 4070. Have clean-installed the newest nvidia drivers (555.85).

**Comment by seerge** [comment]
@Beliroz 

> With AC the dGPU would be off by default when running on battery or USB power, and turn itself on when i started programs like Adobe Bridge (which I use quite frequently). When the dGPU activated it would also show "1 Display running on GPU".

Where are you checking this ?

If you mean automatic GPU enabling when plugged in and disabling when on battery, in both Armory Crate and G-Helper it's called "Optimized" mode. And it behaves exactly same.

> The tray icon for the GPU seems to indicate that the GPU is always activated, even in Eco mode.

In Eco mode GPU is totally powered off. It's not even visible in Device Manager.

--

As for what GPU use for what app - is not anyhow managed by G-Helper or Armoury, but by Nvidia drivers / Optimus and Windows itself.

**Reply by Beliroz** [reply]
> Where are you checking this ?

By hovering over the tray icon for the GPU. It states how many programs and how many displays are running on the GPU.
The tray icon would be greyed out when running on battery and AC's "Silent" mode, indicating the dGPU was inactive. But would activate when starting graphics intensive programs. It would remain activated after shutting down these programs and I had to deactivate the GPU manually in AC.

> In Eco mode GPU is totally powered off. It's not even visible in Device Manager.

Yeah I noticed, the tray icon goes away completely, and does not reappear when swapping to Standard mode. 

When running on any other mode than Ultimate, the display gets disconnected. Programs does not seem to utilize the GPU effectively, even if the tray icon states the programs are running on the GPU.

My main gripe is having to swap to Ultimate mode and restart whenever I want to do photo editing and gaming, and then having to swap it to Optimized and restart to get more than 45 minutes of battery life.

Edit:
I just swapped to Eco mode and back to Standard. This did not reactivate the GPU. The GPU is active upon restart, but the display is still disabled(?). Disconnected? Why does changing GPU mode affect the detection of the display in this way?

**Reply by seerge** [reply]
> When running on any other mode than Ultimate, the display gets disconnected. Programs does not seem to utilize the GPU effectively, even if the tray icon states the programs are running on the GPU.

When you are not in Ultimate (i.e. you are in Standard i assume) your internal display is routed to iGPU by default and completely managed by Nvidia Advanced Optimus . You can check / adjust those settings in Nvidia Control Panel -> Display Mode. Neither AC nor G-Helper is not involved here anyhow.

As for what GPU each app uses is managed by windows, you can set that in Windows Settings -> Graphics

I suggest to make yourself [familiar with readme](https://github.com/seerge/g-helper?tab=readme-ov-file#video_game-gpu-modes)

**Reply by seerge** [reply]
> The GPU is active upon restart, but the display is still disabled(?). 

What kind if "display" do you mean and how do you know it's "disabled" ?

**Reply by Beliroz** [reply]
> What kind if "display" do you mean and how do you know it's "disabled" ?

The display on the laptop itself. I have no external displays atm. I have attached a few screenshots. Left is in standard mode, right is ultimate.

![Untitled-1](https://github.com/seerge/g-helper/assets/170567702/1a37098a-6df4-4515-9ccc-086e088a6492)

**Reply by Beliroz** [reply]
From the Windows Settings. I am fairly certain the display would always be activated/connected with AC regardless of mode. I can reinstall AC tomorrow and double check to make sure I'm not talking out of my ass.

![Untitled-2](https://github.com/seerge/g-helper/assets/170567702/c5dc9156-7065-4add-82a9-962a823258b5)

**Reply by seerge** [reply, marked as answer]
@Beliroz as I have mentioned, when dGPU is enabled (i.e. you have set Standard mode or Optimized and plugged in) - display is managed by ``Nvidia Control Panel -> Display Mode`` . This is called **Nvidia Advanced Optimus**, there you can select if display should be driven by iGPU or dGPU.

From your last screenshot it says that display is connected to "Microsoft Basic Display Driver" . It simply means that **your drivers are not installed properly** or damaged (as windows doesn't use Nvidia driver at all, it uses own fallback driver)

I advice to download latest drivers from [official Nvidia website](https://www.nvidia.com/download/index.aspx) and reinstall them. During that process don't forget to check "Factory reset" in Nvidia installer. Do that in default Standard mode in G-Helper (so GPU is enabled). 

Also install [latest drivers from AMD](https://www.amd.com/en/support/download/drivers.html) for iGPU

**Reply by Beliroz** [reply]
I ran DDU for both the i- and dGPU, reinstalled AC and found out that something had gone terribly wrong with one of the graphics drivers. Now everything is working perfectly, the GPUs are switching on and off, no more weird display deactivation or fallbacks to the Microsoft drivers. Thanks for the help!

**Reply by seerge** [reply]
@Beliroz Np, do I understand correctly that it was also not working as you expected with AC, but after installing drivers now it works fine even without AC as well?

**Reply by Beliroz** [reply]
Yes. I was having the same issues. None of the options in AC would change the behavior of the GPU. All display options in Nvidia control panel were greyed out, so I could not activate Optimus.
I tried reinstalling the dGPU driver multiple times, with using DDU first, in both Ultimate and Standard mode. This did not fix it.
I then tried reinstalling the AMD drivers. The installation wizard failed and sad there was a problem with a graphics driver.
So I ran DDU for the dGPU and iGPU. Then i reinstalled and updated the AMD drivers, followed by a new clean install of the dGPU drivers.
And now it works. The modes in GH works as expected and I could activate Optimus so that the GPU activates and deactivates properly. It still says 0 displays connected, but the options in Windows settings and GH are correct.

**Reply by seerge** [reply]
Ok, i'm glad it's resolved

**Comment by thelastlongbow** [comment]
Hey @seerge! Thanks for the amazing software.

Does Eco (iGPU only) mode stop the HDMI port from working? I know the both the HDMI and right-side USB-C ports are directly connected to the dGPU, but I figured the system would be able to switch to iGPU when on battery. Am I wrong in assuming this?

**Reply by seerge** [reply]
@thelastlongbow hello, when you disable dGPU all ports connected to it obviously won't work.

**Reply by thelastlongbow** [reply]
That makes sense! Sorry I'm just trying to understand the laptop a bit better. Appreciate your reply!
